### PR TITLE
Add support for MySQL configuration

### LINKED ISSUE #2434: documentation on adding azure mysql server parameters in yaml
**Describe the current behavior**
As a part of security , we want to tune mysql server variables and want to be specified in yaml file.
documentation/examples dosent show a way to specify server parameters for mysql Ex:

`apiVersion: azure.microsoft.com/v1alpha1
kind: MySQLDatabase
metadata:
  name: mysqldatabase-sample2
spec:
  resourceGroup: resourcegroup-azure-operators
  server: mysqlserver-sample2`

  
**Describe the improvement**

`apiVersion: azure.microsoft.com/v1alpha1
kind: MySQLDatabase
metadata:
  name: mysqldatabase-sample2
spec:
  resourceGroup: resourcegroup-azure-operators
  server: mysqlserver-sample2
  **sql_mode: STRICT_ALL_TABLES
  local_infile: OFF**`

**Additional context**

we currently use azure portal or GUI , after deploy to change these variables, but goal is to specify in yaml if possible while deploying

### PR DESCRIPTION
**Pull request by matthchr**
Also updates MySQL Administrator to use the new 20220101 GA API rather than the preview API.

Closes #2434

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

File: docs/hugo/content/reference/_index.md [hunk 1]
```diff
@@ -155,17 +155,18 @@ Azure Database for MySQL - Single Server is on the retirement path and is [sched
...
+| [FlexibleServer](https://azure.github.io/azure-service-operator/reference/dbformysql/v1api20210501/#dbformysql.azure.com/v1api20210501.FlexibleServer)                             | 2021-05-01  | v1api20210501  | v2.0.0         | [View](https://github.com/Azure/azure-service-operator/tree/main/v2/samples/dbformysql/v1api/v1api20210501_flexibleserver.yaml)                |
+| [FlexibleServer](https://azure.github.io/azure-service-operator/reference/dbformysql/v1beta20210501/#dbformysql.azure.com/v1beta20210501.FlexibleServer)                           | 2021-05-01  | v1beta20210501 | v2.0.0-beta.0  | [View](https://github.com/Azure/azure-service-operator/tree/main/v2/samples/dbformysql/v1beta/v1beta20210501_flexibleserver.yaml)              |
+| FlexibleServersAdministrator                                                                                                                                                       | 2022-01-01  | v1api20220101  | v2.1.0         | -                                                                                                                                              |
```
Comment (theunrepentantgeek) [inline docs/hugo/content/reference/_index.md:164]: Both the link to the docs and the link to the sample have gone

Comment (theunrepentantgeek) [reply docs/hugo/content/reference/_index.md:164]: Oh, you've dropped the preview version. Breaking change?

File: v2/azure-arm.yaml [hunk 2]
```diff
@@ -936,17 +931,20 @@ objectModelConfiguration:
       FlexibleServers_FirewallRule:
         $exportAs: FlexibleServersFirewallRule
         $supportedFrom: v2.0.0-alpha.2
-    2021-12-01-preview:
+    2022-01-01:
```
Comment (theunrepentantgeek) [inline v2/azure-arm.yaml:934]: Do we want this breaking change?

File: v2/azure-arm.yaml [hunk 3]
```diff
@@ -936,17 +931,20 @@ objectModelConfiguration:
...
       FlexibleServers_Administrator:
         $exportAs: FlexibleServersAdministrator
-        $supportedFrom: v2.0.0
+        $supportedFrom: v2.1.0
```
Comment (theunrepentantgeek) [inline v2/azure-arm.yaml:937]: Looks like we released this in v2.0.0, probably need to retain it in v2.1.0

Comment (theunrepentantgeek) [reply v2/azure-arm.yaml:937]: Unless we don't get a choice because the underlying OpenAPI schema has gone.

**Review by theunrepentantgeek — APPROVED** [review]
Seems odd to be removing a resource we only just released, but otherwise looks Ok.

File: docs/hugo/content/reference/_index.md (hunk 1 above)
Comment (matthchr) [reply docs/hugo/content/reference/_index.md:164]: No because it hasn't been released yet.

File: v2/azure-arm.yaml (hunk 2 above)
Comment (matthchr) [reply v2/azure-arm.yaml:934]: Hasn't been released yet, so safe. I only just merged support for it like a week ago.

File: v2/azure-arm.yaml (hunk 3 above)
Comment (matthchr) [reply v2/azure-arm.yaml:937]: That was a copy-paste mistake. It's actually 2.1.0 (which is why it's safe for me to remove)

Comment (matthchr) [reply v2/azure-arm.yaml:937]: For completeness, it was in #2954, which was only merged 5 days ago.

Comment (theunrepentantgeek) [reply v2/azure-arm.yaml:937]: That works. 👍

**Pull request merged**